version: "3.9"
services:
  localstack:
    container_name: localstack-for-rds-bug
    image: localstack/localstack-pro:3.0.2
    ports:
      - "443:443"
      - "4566:4566"
      - "4510-4559:4510-4559/tcp"
    environment:
      - RDS_PG_CUSTOM_VERSIONS=14.3
      - SERVICES=s3,sqs,sns,secretsmanager,rds,sts,ec2,cloudformation,ecr,ssm,iam,events
      - DEBUG=1
      - LS_LOG=trace
      - DOCKER_HOST=unix:///var/run/docker.sock
      - LOCALSTACK_AUTH_TOKEN=PUT_YOUR_TOKEN_HERE
    volumes:
      - ./:/var/lib/localstack/rds_bug
      - ./ready-scripts:/etc/localstack/init/ready.d
      - /var/run/docker.sock:/var/run/docker.sock
    healthcheck:
      test: ["CMD-SHELL", "awslocal rds wait db-instance-available --db-instance-identifier my-db --cli-read-timeout=1 --cli-connect-timeout=1"]
      interval: 10s
      timeout: 10s
      retries: 30

  migrate-db:
    image: migrate/migrate:4
    container_name: migrate-db-for-rds-bug
    volumes:
      - ./migrations:/migrations
    command: --path /migrations -database postgres://dummyuser:dummypassword@localstack:4511/mydatabase?sslmode=disable up
    depends_on:
      localstack:
        condition: service_healthy
